packages/core/lib/get-graphql-parameters.ts

```diff
@@ -1,26 +1,60 @@
-import { isHttpMethod } from "./util/index";
-import { GraphQLParams, Request } from "./types";
+import { GraphQLParams } from "./types";
 
-export const getGraphQLParameters = (request: Request): GraphQLParams => {
-  const { body, method, query: queryParams } = request;
+function setValue(obj: any, path: string, value: any) {
+  const a = path.split(".");
+  let o = obj;
+  while (a.length - 1) {
+    const n = a.shift()!;
+    if (!(n in o)) o[n] = {};
+    o = o[n];
+  }
+  o[a[0]] = value;
+}
+
+export async function getGraphQLParameters(request: Request): Promise<GraphQLParams> {
+  const url = new URL(request.url);
+
+  let operationName: string | undefined;
+  let query: string | undefined;
+  let variables: any | undefined;
 
-  let operationName;
-  let query;
-  let variables;
+  switch (request.method) {
+    case "GET": {
+      operationName = url.searchParams.get("operationName") || undefined;
+      query = url.searchParams.get("query") || undefined;
+      variables = url.searchParams.get("variables") || undefined;
+      break;
+    }
```

Comment:
can the post and get avenues be split? Sometimes the GET routes one doesnt want supported.

Reply:
Isn't that something user can block in their server impl?

Reply:
they can, but then there is 5 lines of code shipped doing absolutely nothing. Its simple;

```
getGETHandler();
getPOSTHandler();
gethandleReqRes(); // that calls both.
```

Reply:
Helix isnt about making decisions for you, its a "build your own server" library. Making the decision that GET/POST is supported isnt something that is in scope. All the user to "build their own server". Helix should just handle the query in and execution layers, leave the rest up to helpers.

Reply:
I think separating get/post will break SSE for subscriptions, since it's using GET for the Event-Stream. 

See https://stackoverflow.com/questions/34261928/can-server-sent-events-sse-with-eventsource-pass-parameter-by-post

@n1ru4l please confirm?

Reply:
I generally would still expose that as 2 methods, as some folk won't be using sse. Export and consume what you're going to use.

Reply:
Actually SSE can be done with both POST and GET. I have been thinking for a while how we could make every functionality of helix more modular.

The HTTP verb specifies how the GraphQL payload sent by the client should be parsed. So on the server we could have some kind of pattern for parsing that:

```ts
const payload = await parsePayload({
  POST: postParser,
  GET: getParser
})(incomingRequest)
```

if people want custom behavior or non standard stuff they can simple replace the handler for a given method or remove it.

This approach is more verbose in boilerplate but a bit more modular in giving. A higher level API such as yoga can then provide a opinionated API.

The response protocol is then identified via the content-type and/or accept header, but also the operation type. Today the specification is pretty vague about this, I opened https://github.com/graphql/graphql-over-http/issues/167 a while ago.

We can have a similar API to how we parse the payload for the response protocol:

```ts
const response = await matchResponse([
  { match: isSSE ,handler: sseResponseBuilder },
  { match: isMultiPart, handler: multiPartResponseBuilder },
  { match: isApplicationGraphQL, handler: applicationGraphQLResponseBuilder },
  // required inputs for determining the response
])(payload, req.headers, req.contentType, streamOrSinglePayloadFromExecute)
```

Reply:
@maraisr You can prevent it to parse GET requests by a simple condition. You're still in charge here to decide when the request should be parsed.
```
if (request.method !== 'GET') {
  const params = await getGraphQLParameters(request);
}
```

Reply:
6 lines of code included that won't be used. Death by a thousand cuts. Think optimally.